src/fsharp/CompileOps.fs

```diff
@@ -5085,59 +5140,112 @@ module private ScriptPreprocessClosure =
+                                        tcConfigB.packageManagerLines <- tcConfigB.packageManagerLines |> Map.map(fun _ l -> l |> List.filter(fun (tried, _, _) -> tried))
+                                        tcConfig := TcConfig.Create(tcConfigB, validate=false)]
+            else []
+
+        and loop (ClosureSource(filename, m, sourceText, parseRequired)) = 
+            [   if not (observedSources.HaveSeen(filename)) then
                     observedSources.SetSeen(filename)
                     //printfn "visiting %s" filename
                     if IsScript filename || parseRequired then 
                         let parseResult, parseDiagnostics =
-                            let errorLogger = CapturingErrorLogger("FindClosureParse")                    
+                            let errorLogger = CapturingErrorLogger("FindClosureParse")
                             use _unwindEL = PushErrorLoggerPhaseUntilUnwind (fun _ -> errorLogger)
                             let result = ParseScriptText (filename, sourceText, !tcConfig, codeContext, lexResourceManager, errorLogger) 
                             result, errorLogger.Diagnostics
 
                         match parseResult with 
-                        | Some parsedScriptAst ->                    
-                            let errorLogger = CapturingErrorLogger("FindClosureMetaCommands")                    
+                        | Some parsedScriptAst ->
+                            let errorLogger = CapturingErrorLogger("FindClosureMetaCommands")
                             use _unwindEL = PushErrorLoggerPhaseUntilUnwind (fun _ -> errorLogger)
-                            let pathOfMetaCommandSource = Path.GetDirectoryName filename
+                            let pathOfMetaCommandSource = Path.GetDirectoryName(filename)
+
                             let preSources = (!tcConfig).GetAvailableLoadedSources()
 
                             let tcConfigResult, noWarns = ApplyMetaCommandsFromInputToTcConfigAndGatherNoWarn (!tcConfig, parsedScriptAst, pathOfMetaCommandSource)
                             tcConfig := tcConfigResult // We accumulate the tcConfig in order to collect assembly references
-                        
+
+                            yield! resolveDependencyManagerSources filename
+
                             let postSources = (!tcConfig).GetAvailableLoadedSources()
                             let sources = if preSources.Length < postSources.Length then postSources.[preSources.Length..] else []
 
-                            //for (_, subFile) in sources do
-                            //   printfn "visiting %s - has subsource of %s " filename subFile
-
+                            yield! resolveDependencyManagerSources filename
+#if DEBUG
+                            for (_,subFile) in sources do
+                               printfn "visiting %s - has subsource of %s " filename subFile
```

Comment:
This and the other conditional enclosed printfn below fails the "load-test" test when run locally (in debug mode by default), could it be removed from master?

Reply:
I think the answer to this is to adjust testing all-up, since these kinds of diagnostics are very important for environments like Jupyter (where this is enabled) to see what went wrong

Reply:
Maybe a `#if JUPYTER_NOTEBOOK_DEBUG` is a good approach then? Not a big issue but wanted to report it may be an overlooked remnant.

The diagnostics could also be incorporated in the dependency manager itself (assuming jupyter will have its own) rather than CompileOps.fs.

Reply:
@KevinRansom @dsyme was pointing at the printf here ^^^